firebase-firestore/src/main/java/com/google/firebase/firestore/remote/RemoteStore.java

```diff
@@ -613,9 +638,26 @@ private void startWriteStream() {
     hardAssert(
         shouldStartWriteStream(),
         "startWriteStream() called when shouldStartWriteStream() is false.");
+
+    initializeSslIfNeeded();
     writeStream.start();
   }
 
+  /** Waits for the SSL Provider to be installed. Logs a warning on failure. */
+  private void initializeSslIfNeeded() {
+    try {
+      sslBarrier.await();
+    } catch (InterruptedException e) {
```

Comment:
Who interrupts the thread to generate this exception?

Normally interrupts mean that the system is trying to kill a thread. We shouldn't swallow this interrupt as we're doing here.

Reply:
My inclination here is deal with the error handling as part of b/132978871. My goal is to make a best effort at providing an updated SSL Cipher, but if I can't then this is only an error condition if the SDK fails to connect. That error handling will come in a follow-up and will be part of the stream handling rather than Remote Store.